### PR TITLE
fix deepspeed issue on trainer_rm.py, add crossentropy support

### PR DESCRIPTION
**Pull request by theblackcat102**
# How to use augmented results [#2265](https://github.com/LAION-AI/Open-Assistant/issues/2265)

1. Generate bad responses

* sample using code under tools/augment_oasst.py, please modify the model_name for different models for sample different bad samples. Note the sampling would took a few days to finish.

* Once all different samples are generated, use the AggregateResults class to merge multiple bad response into one dataset and move it under the cache_dir

2. Use the augmented results as replacement for oasst dataset

```
oasst_export_augmented:
  datasets:
    - augment_oasst:
        input_file_path: augmented_latin_cyrillic_oasst_2023-03-27.jsonl
```

# Other contribution of this branch

I was training some baseline for RM models and found some issues running different models (deberta, electra, bloom) under deepspeed

* fix summarized from human feedback (hf_summary) and webgpt dataset for rm, sft and rl mode. Note the implementation of hfsummary under rank_datasets.py is slightly wrong.

* deepspeed distributed gradient issue fixed in trainer_rm.py

* add crossentropy loss support for new code `RMCLSLoss`, mainly because WebGPT seems to use this instead of rankloss and part of the old code migration. (see wandb reward-model search for pythia-410m )

* added pretrained model support for non pythia model for RM training. ( see utils.py get_model )

* and I was having gradient unscale error by setting dtype=fp16 on model initialization as mentioned by dimitri in discord ([link](https://discuss.pytorch.org/t/valueerror-attemting-to-unscale-fp16-gradients/81372)). 

@andreaskoepf I also wanted to discuss about the way how trainer code (trainer_sft.py trainer_rm.py) are executed now. Cause its placed under `model/model_training/` and import modules from under `model_training`. So the problem is how do you execute the code?

1. prepend system path? and run `deepspeed trainer_sft.py ...` under model_training?

2. `python -m model_training.trainer_rm ...` works but doesn't support deepspeed

3. move trainer code under `model`

I am raising this issue mainly because there isn't any instruction on how to execute the trainer in under the model_training/readme

### CODE REVIEW

File: model/model_training/trainer_rm.py [hunk 1]
```diff
@@ -158,7 +160,7 @@ def argument_parsing(notebook=False, notebook_args=None):
     conf["wandb_entity"] = args.wandb_entity
     conf["local_rank"] = args.local_rank
     conf["deepspeed"] = args.deepspeed
-    if args.rng_seed is not None:
+    if "rng_seed" not in conf and args.rng_seed is not None:
```
Comment (andreaskoepf) [inline model/model_training/trainer_rm.py:163]: I don't think this logic is correct, but I can correct it after merge in the improve_small_rm branch..

File: model/model_training/utils.py [hunk 2]
```diff
@@ -296,12 +299,14 @@ def get_model(conf, tokenizer, pad_vocab_size_to_multiple_of=16):
 
     if conf.is_reward_model:
         if "pythia" in conf.model_name:
-            model = GPTNeoXRewardModel.from_pretrained(conf.model_name, cache_dir=conf.cache_dir, torch_dtype=dtype)
```
Comment (andreaskoepf) [inline model/model_training/utils.py:299]: hm. why was the dtype specification here removed .. ?

**Review by andreaskoepf — APPROVED** [review]
ok, very cool! thank you.

File: model/model_training/trainer_rm.py (hunk 1 above)
Comment (theblackcat102) [reply model/model_training/trainer_rm.py]: its my bad, I have reverted back

File: model/model_training/utils.py (hunk 2 above)
Comment (theblackcat102) [reply model/model_training/utils.py]: mainly due to unable to rescale fp16, but as you mentioned, we could just assign fp32 for that
